lib/parser.js

```diff
@@ -128,14 +128,31 @@ Parser.prototype._parsePair = function(str) {
  */
 
 Parser.prototype._parseVal = function(node) {
-  node.val = node.str.replace(Parser.QUOTES, '$2');
+  if (typeof node.str === 'string') {
+    node.val = node.str.replace(Parser.QUOTES, '$2');
+  }
 
-  return this._parseBool(node)
+  return this._parseUndefined(node)
+    || this._parseBool(node)
     || this._parseNum(node)
     || this._parseStr(node);
 };
 
 /**
+ * Parse undefined value.
+ *
+ * @param {Object} node
+ * @returns {Object}
+ * @private
+ */
+
+Parser.prototype._parseUndefined = function(node) {
+  if (node.val === void 0) {
```

Comment:
LOL @n3bulous 

It's just another way to say `undefined`. It used to be promoted as a way to be safe against those that might overwrite `undefined` (yes, possible in non-strict mode); however, not something people worry too much about these days. At the same time, I have no problem with it when used. Not to mention, it's shorter than `undefined`. Not that it matters, but I guess that's a benefit :smile: